rust-toolchain.toml

```diff
@@ -0,0 +1,4 @@
+[toolchain]
```

Comment:
If we have a `rust-toolchain.toml` file, it would be good to do an explicit install step instead of "implying" this with the first call to cargo. Reason being that the cache action works out the cache key based on the [installed rust version](https://github.com/Swatinem/rust-cache#cache-details) and that would happen before the first cargo invocation, effectively creating the wrong cache key for what is actually being used.

See https://github.com/actions-rs/toolchain/issues/126 + recommended solution.